### PR TITLE
Add additional span to IncorrectValue error

### PR DESCRIPTION
**Pull request by kubouch**
# Description

Fixes the IncorrectValue ShellError to point at the command which caused the error (otherwise it's hard to find out where the error was triggered).

Before:
```
> "C:\\Users\\kubouch" | str replace $env.HOME '~'
Error: nu::shell::incorrect_value

  × Incorrect value.
    ╭─[Host Environment Variables:10:1]
 10 │ "GITDIR"="C:\\Users\\kubouch\\git"
 11 │ "HOME"="C:\\Users\\kubouch"
    ·        ──────────┬─────────
    ·                  ╰── Regex error: Parsing error at position 4: Invalid hex escape
 12 │ "HOMEDRIVE"="C:"
    ╰────

>  "22688-04-11T23:47:17" | into datetime | into int
Error: nu::shell::incorrect_value

  × Incorrect value.
   ╭─[entry #9:1:1]
 1 │ "22688-04-11T23:47:17" | into datetime | into int
   ·                                          ────┬───
   ·                                              ╰── DateTime out of range for timestamp: 1677-09-21T00:12:43Z to 2262-04-11T23:47:16
   ╰────
```

After:
```
> "C:\\Users\\kubouch" | str replace $env.HOME '~'
Error: nu::shell::incorrect_value

  × Incorrect value.
    ╭─[Host Environment Variables:42:1]
 42 │ "GOPATH"="C:\\Users\\kubouch\\.go"
 43 │ "HOME"="C:\\Users\\kubouch"
    ·        ──────────┬─────────
    ·                  ╰── Regex error: Parsing error at position 4: Invalid hex escape
 44 │ "HOMEDRIVE"="C:"
    ╰────
   ╭─[entry #21:1:1]
 1 │ "C:\\Users\\kubouch" | str replace $env.HOME '~'
   ·                        ─────┬─────
   ·                             ╰── encountered here
   ╰────

> "22688-04-11T23:47:17" | into datetime | into int
Error: nu::shell::incorrect_value

  × Incorrect value.
   ╭─[entry #2:1:1]
 1 │ "22688-04-11T23:47:17" | into datetime | into int
   · ───────────┬──────────                   ────┬───
   ·            │                                 ╰── encountered here
   ·            ╰── DateTime out of range for timestamp: 1677-09-21T00:12:43Z to 2262-04-11T23:47:16
   ╰────

```

# User-Facing Changes

Better error message

# Tests + Formatting

# After Submitting

### CODE REVIEW

File: crates/nu-protocol/src/shell_error.rs [hunk 1]
```diff
@@ -115,6 +115,8 @@ pub enum ShellError {
         msg: String,
         #[label = "{msg}"]
         span: Span,
+        #[label = "encountered here"]
+        orig_span: Span,
```
Comment (sholderbach) [inline crates/nu-protocol/src/shell_error.rs:119]: Mhh I think the span names are a bit confusing. To me `orig_span` sounds more like the origin of a Value.

**Review by sholderbach — COMMENTED** [review]
Thanks this looks much better.

Although we are not yet consistent with how we do it across `ShellError` not sure about how best to name the spans.

**Review by amtoine — COMMENTED** [review]
if we address the thread from @sholderbach, i think this looks much better too from what i see in the PR description, thanks @kubouch :pray:

File: crates/nu-protocol/src/shell_error.rs (hunk 1 above)
Comment (kubouch) [reply crates/nu-protocol/src/shell_error.rs]: @sholderbach Better now?

Comment (amtoine) [reply crates/nu-protocol/src/shell_error.rs]: (done in https://github.com/nushell/nushell/pull/10036/commits/00b27604b38b16b35b0e97f9bdcf387b7a2be867)

**Comment by kubouch** [comment]
@sholderbach If the error fields are still cofusing, let me know.

**Comment by sholderbach** [comment]
Thanks! The new names are perfect!

**Pull request merged**